Move Heat Exchanger implementation to ImmersiveTech

ThePatmann opened the pull request:
This solution implements Heat Exchanger and Condensation Tube recipes in the Immersive Technology Heat Exchanger and Radiator respectively.
It also includes the Solar Tower power change that was committed while I was working on this.
The Solar Tower's Molten Sodium is treated like a NaK Coolant in regards to the Heat Exchanger recipes (with some caveats to make it act like a simplified tutorial for the Nuclearcraft Reactor Coolant loops, while still allowing for simple and fast power set-ups in case the user doesn't wish to go all the way into multi-turbine power generation)

ThePatmann committed:
Squash Steam-Types & Implement Radiator Condensation

Distiller Recipe Tweaked to use Condensate as I am phasing out Distilled Water (which is categorically identical)
Solar Tower now takes Condensate in appropriate amounts, it's output is the same.
High Pressure Steam Turbine now outputs correct type of exhaust steam
Radiator acts as a Nuclearcraft Condenser pending actual implementation of Condensation Tubes
TODO: Implement Heat Exchanger Recipes

ThePatmann committed:
Move Heat Exchanger Recipes

The Chemical Reactor will no longer be used for Heat Exchange activities

ThePatmann committed:
Transfer Heat Exchanger Recipes

Implements Heat Exchanger using Immersive Technology instead of Chemical Reactor
Allows Molten Sodium from ImmTech Solar Tower to be used as a coolant for the Heat Exchanger a la Nuclearcraft (Some recipes are tenative and perhaps too slow for pracitical usage, further testing required)
Removes Condensator recipes from Fluid Infuser
Updates Solar Tower Power Output (4x Steam Turbine at max speed, as opposed to 1/4 at minimum speed)

ThePatmann committed:
Merge branch 'master' into master

ThePatmann commented:
Wait, There's a problem with line 67 68 and 69 in the Steam.zs file. Make sure they have semicolons at the end of them before committing.

EastAPOLO commented on scripts/steam.zs:
```diff
@@ -1,33 +1,134 @@
+mods.immersivetechnology.HeatExchanger.addRecipe(<liquid:moltensodium> * 1, <liquid:low_pressure_steam> * 4608,  <liquid:superheatedmoltensodium> * 1, <liquid:exhaust_steam> * 4608, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<liquid:moltensodium> * 25, <liquid:super_critical_steam> * 192,  <liquid:superheatedmoltensodium> * 25, <liquid:well_pressed_water> * 192, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<liquid:moltensodium> * 1, <liquid:high_pressure_steam> * 432,  <liquid:superheatedmoltensodium> * 1, <liquid:crap_steam> * 432, 0, 1);
+//mods.immersivetechnology.HeatExchanger.addRecipe(outSide, outTop,  inSide, inTop, 0, 1); //mnemonic reminder
```
Please remove the commented lines.

ThePatmann replied:
Are you sure about that? I left those comments in as alternatives in case you were not comfortable with the current recipes.

EastAPOLO replied:
I think going with the nuclearcraft recipe is a good choice because I want the power generation in the modpack to be compatible with nuclearcraft as much as possible. We can always tweak this more in the future if anything needs changing.

ThePatmann committed:
Added Semicolons

EastAPOLO commented on scripts/steam.zs:
```diff
@@ -1,33 +1,134 @@
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:corium_nak>*60, <fluid:condensate_water>*125, <fluid:corium_nak_hot>*60, <fluid:preheated_water>*125, 0, 1)
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*5120, <fluid:high_pressure_steam>*672, <fluid:super_critical_steam>*80, <fluid:crap_steam>*672, 0, 1)
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*320, <fluid:low_pressure_steam>*448, <fluid:super_critical_steam>*5, <fluid:exhaust_steam>*448, 0, 1)
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*1280, <fluid:preheated_water>*224, <fluid:super_critical_steam>*20, <fluid:condensate_water>*224, 0, 1)
```
These lines require a semicolon.

EastAPOLO reviewed (changes requested):
I left some feedback that needs to be addressed, overall looks good.

ThePatmann committed:
Remove Autogenerated Files

Installed Nuclearcraft Addons generate these files every launch
This removes this pointless stored directory

EastAPOLO commented on scripts/steam.zs:
```diff
@@ -1,33 +1,134 @@
+	}
+}
+
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<liquid:corium_nak_hot> * 60, <liquid:preheated_water> * 125);
+mods.immersivetechnology.HeatExchanger.addRecipe(<liquid:corium_nak>*60, <liquid:condensate_water>*125, <liquid:corium_nak_hot>*60, <liquid:preheated_water>*125, 0, 1);
```
These lines appear to be duplicates and should be removed. Also that would prevent an error with the chemical reactor since it is trying to remove a recipe twice.

EastAPOLO commented on scripts/steam.zs:
```diff
@@ -1,33 +1,134 @@
 mods.immersivetechnology.Boiler.removeFuel(<liquid:biodiesel>);
 mods.immersivetechnology.Boiler.removeFuel(<liquid:diesel>);
 mods.immersivetechnology.Boiler.removeFuel(<liquid:gasoline>);
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<fluid:corium_nak_hot>*60, <fluid:preheated_water>*125);
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<fluid:super_critical_steam>*80, <fluid:crap_steam>*672);
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<fluid:super_critical_steam>*5, <fluid:exhaust_steam>*448);
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<fluid:super_critical_steam>*20, <fluid:condensate_water>*224);
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<fluid:high_pressure_steam>*75, <fluid:exhaust_steam>*120);
+mods.nuclearcraft.ChemicalReactor.removeRecipeWithInput(<fluid:high_pressure_steam>*80, <fluid:condensate_water>*16);
+mods.nuclearcraft.Enricher.removeRecipeWithInput(<minecraft:snowball>, <fluid:exhaust_steam>*16000);
+mods.nuclearcraft.Enricher.removeRecipeWithInput(<minecraft:snowball>, <fluid:low_quality_steam>*16000);
```
I think we could replace fluid with liquid here?

EastAPOLO commented on scripts/steam.zs:
```diff
@@ -1,33 +1,134 @@
 mods.immersivetechnology.PressurizedFluid.add(<fluid:distwater>);
 mods.immersivetechnology.PressurizedFluid.add(<fluid:high_pressure_steam>);
+
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*300, <fluid:low_pressure_steam>*120, <fluid:high_pressure_steam>*75, <fluid:exhaust_steam>*120, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*320, <fluid:preheated_water>*16, <fluid:high_pressure_steam>*80, <fluid:condensate_water>*16, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:corium_nak>*60, <fluid:condensate_water>*125, <fluid:corium_nak_hot>*60, <fluid:preheated_water>*125, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*5120, <fluid:high_pressure_steam>*672, <fluid:super_critical_steam>*80, <fluid:crap_steam>*672, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*320, <fluid:low_pressure_steam>*448, <fluid:super_critical_steam>*5, <fluid:exhaust_steam>*448, 0, 1);
+mods.immersivetechnology.HeatExchanger.addRecipe(<fluid:low_pressure_steam>*1280, <fluid:preheated_water>*224, <fluid:super_critical_steam>*20, <fluid:condensate_water>*224, 0, 1);
```
Replacing fluid with liquid should work here just to be consistent with the other heat exchanger recipes.

ThePatmann committed:
Cleanup

Removed excessive Corium heat exchanger removal
Changed from Fluid Registrar to Liquid

ThePatmann committed:
Removed all commented code

Because it won't get merged otherwise. I guess.

ThePatmann committed:
There, now just commit it already
